### DISCUSSION TITLE
Errors in installing RACCOON

### DISCUSSION THREAD

**Discussion by Bigdogcat** [idaholab/moose#28219] [Q&A Modules: Phase field]
Hi all, i followed the steps given on https://hugary1995.github.io/raccoon/getting_started/install.html
But i am getting some errors while executing the 'make' command:

/Users/sunqingpei/projects/raccoon/moose/framework/contrib/hit/src/hit/parse.cc:147:8: error: no member named 'set_type' in 'wasp::HITNodeView'
  _hnv.set_type(wasp::BLANK_LINE);
  ~~~~ ^
/Users/sunqingpei/projects/raccoon/moose/framework/contrib/hit/src/hit/parse.cc:147:23: error: no member named 'BLANK_LINE' in namespace 'wasp'
  _hnv.set_type(wasp::BLANK_LINE);
                ~~~~~~^
/Users/sunqingpei/projects/raccoon/moose/framework/contrib/hit/src/hit/parse.cc:295:28: error: no member named 'BLANK_LINE' in namespace 'wasp'
  if (_hnv.type() == wasp::BLANK_LINE)
                     ~~~~~~^
Really appreciate the help!

**Comment by brandonlangley** [comment]
@Bigdogcat -

This indicates that your of WASP version is out of date with your MOOSE version, and these should be in sync.

That `set_type()` method and `BLANK_LINE` enum were added to WASP around 10 months ago on Oct 6, 2023.

I do not see anything on https://hugary1995.github.io/raccoon/getting_started/install.html with updating WASP.

I am not sure if you used those https://hugary1995.github.io/raccoon/getting_started/conda.html Conda steps.

But it appears that the `moose-dev` Conda package (which is needed for WASP) is not mentioned on that page.

You would need something like https://mooseframework.inl.gov/getting_started/installation/conda.html for that.

If you are not using that Conda setup, then `./moose/scripts/update_and_rebuild_wasp.sh` will need to be ran.

This is in addition to running the `update_and_rebuild_petsc.sh` and `update_and_rebuild_libmesh.sh` scripts.

It may be best to wait for @hugary1995 to respond with more insight on which of the setups is recommended.

Because the `moose-dev=2024.07.19` version at that link above might not be sync with your RACCOON version.

@hugary1995 - Do you have any recommendations about this?

**Reply by hugary1995** [reply]
The RACCOON installation guide is outdated. I'm running out of bandwidth for maintaining it, and I am looking for the next maintainer to hand it over. If I don't find one I might have to archive it.

**Reply by Bigdogcat** [reply]
Thanks very much for the reply. I reinstall MOOSE as you suggested and compile RACCOON again, but there are still some errors as follows, could you please help have a look

/Users/sunqingpei/projects/raccoon/moose/framework/contrib/tinyhttp/src/tinyhttp/http.C:28:37: error: use of undeclared identifier 'MSG_NOSIGNAL'
    if (::send(mSocket, what, size, MSG_NOSIGNAL) < 0)
                                    ^
/Users/sunqingpei/projects/raccoon/moose/framework/contrib/tinyhttp/src/tinyhttp/http.C:35:43: error: use of undeclared identifier 'MSG_NOSIGNAL'
    if ((len = recv(mSocket, target, max, MSG_NOSIGNAL)) < 0)
                                          ^
/Users/sunqingpei/projects/raccoon/moose/framework/contrib/tinyhttp/src/tinyhttp/http.C:46:35: error: use of undeclared identifier 'MSG_NOSIGNAL'
        if (recv(mSocket, &ch, 1, MSG_NOSIGNAL) != 1)
                                  ^

**Reply by Bigdogcat** [reply]
Resolved by referring to https://github.com/idaholab/moose/discussions/27489#discussion-6566338

**Reply by Bigdogcat** [reply]
It's a little bit embarrassing that I have resolved this tinyhttp error by referring to https://github.com/idaholab/moose/discussions/27489#discussion-6566338. But when I compile, there is still an error
```
ld: warning: directory not found for option '-L/Users/sunqingpei/miniconda3/envs/moose/lib/gcc/x86_64-apple-darwin13.4.0/9.5.0'
ld: library not found for -lvtkParallelMPI-9.2
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Reply by milljm** [reply]
Deleting original response in favor of answering https://github.com/idaholab/moose/discussions/28224#discussioncomment-10174020
